presto-main/src/main/java/com/facebook/presto/server/remotetask/ReactorNettyHttp2Client.java

```
@@ -0,0 +1,333 @@
+    private static final HeaderName CONTENT_TYPE_HEADER_NAME = HeaderName.of("Content-Type");
+    private static final HeaderName CONTENT_LENGTH_HEADER_NAME = HeaderName.of("Content-Length");
+
+    private final HttpClient httpClient;
+
+    @Inject
+    public ReactorNettyHttp2Client(ReactorNettyHttp2ClientConfig config)
+    {
+        SslContext sslContext;
+        try {
+            File keyFile = new File(config.getKeyStorePath());
+            File trustCertificateFile = new File(config.getTrustStorePath());
+            PrivateKey privateKey = loadPrivateKey(keyFile, Optional.of(config.getKeyStorePassword()));
+            X509Certificate[] certificateChain = readCertificateChain(keyFile).toArray(new X509Certificate[0]);
+            X509Certificate[] trustChain = readCertificateChain(trustCertificateFile).toArray(new X509Certificate[0]);
+
+            SslProvider provider = SslProvider.isAlpnSupported(SslProvider.OPENSSL) ? SslProvider.OPENSSL : SslProvider.JDK;
+            SslContextBuilder sslContextBuilder = SslContextBuilder.forClient()
+                    .sslProvider(provider)
+                    .protocols("TLSv1.2")
+                    .keyManager(privateKey, certificateChain)
+                    .trustManager(trustChain)
+                    .applicationProtocolConfig(new ApplicationProtocolConfig(ALPN, NO_ADVERTISE, ACCEPT, HTTP_2, HTTP_1_1));
+            if (config.getCipherSuites().isPresent()) {
+                sslContextBuilder.ciphers(Splitter
+                        .on(',')
+                        .trimResults()
+                        .omitEmptyStrings()
+                        .splitToList(config.getCipherSuites().get()));
+            }
+
+            sslContext = sslContextBuilder.build();
+        }
+        catch (IOException | GeneralSecurityException e) {
+            throw new RuntimeException("Failed to configure SSL context", e);
+        }
+
+        if (!Epoll.isAvailable()) {
```

Maybe fail if epoll is not available on Linux. Consider adding unavailability cause to the failure